win32ss/user/winsrv/consrv/shutdown.c

```diff
@@ -23,15 +55,41 @@ ConsoleClientShutdown(IN PCSR_PROCESS CsrProcess,
                       IN BOOLEAN FirstPhase)
 {
     PCONSOLE_PROCESS_DATA ProcessData = ConsoleGetPerProcessData(CsrProcess);
+    NTSTATUS Status;
+
+    /* Do not kill system processes when a user is logging off */
+    if ((Flags & EWX_SHUTDOWN) == EWX_LOGOFF &&
+        (CsrProcess->ShutdownFlags & (CsrShutdownSystem | CsrShutdownOther)))
+    {
+        DPRINT("Do not kill a system process in a logoff request!\n");
+        return CsrShutdownNonCsrProcess;
+    }
 
-    //FIXME: UNIMPLEMENTED!
+    /* Do not kill Winlogon or CSRSS */
+    if (CsrProcess->ClientId.UniqueProcess == NtCurrentProcess() ||
```

Comment:
This doesn't seem like it will work. `NtCurrentProcess()` returns the constant `(HANDLE)-1` , whereas `UniqueProcess` probably contains the actual process id.
`NtCurrentTeb()->ClientId.UniqueProcess` or `GetCurrentProcessId()`?

Reply:
indeed. and usersrv has the same bug

Reply:
That's definitively a bug and I confirm we need to use the NtCurrentTeb() solution.